module_info_partner/models/module_partner.py

```diff
@@ -30,17 +49,26 @@ def _prepare_partner_module_vals(self, partner, version, module, module_info):
 
     @api.model
     def update_or_create(self, partner, version_num, module_info):
-        tech_name = module_info.get("name")
         # check if module already exists
         module_info_obj = self.env["module.information"]
-        module = module_info_obj.search([("technical_name", "=", tech_name)])
+
+        partner_id = partner.id if module_info["is_custom"] else False
+        module = module_info_obj.search(
+            [
+                ("name", "=", module_info["name"]),
+                ("partner_id", "=", partner_id),
+            ]
+        )
         if not module:
-            module = module_info_obj.create(self._prepare_module_info_vals(module_info))
-        version = self.env["odoo.version"].search([("version", "=", version_num)])
+            # in this case, Module is custom
+            # create it with the customer partner_id
+            module = module_info_obj.create(
+                self._prepare_module_info_vals(module_info, partner)
+            )
+        version = self.env["odoo.version"].search([("name", "=", version_num)])
```

Comment:
this field should be protected if there is no version_num already registered (like v17) , it fails at the SQL INSERT statment.

So, either improve the error message, or allow to create new version here and take care to handle  how master version